ebmlite/core.py

```diff
@@ -811,6 +811,24 @@ def __repr__(self):
         return "<%s %r at 0x%08X>" % (self.__class__.__name__, self.name,
                                       id(self))
 
+    def __enter__(self):
+        """
+        Enter context manager for this document.
+
+        Expects to clean up `self.stream` object; raises an error if stream
+        object needs no such cleanup.
+        """
+        if not hasattr(self.stream, 'close'):
+            raise TypeError(
+                "invalid stream object for context management"
+                " (expected a stream object with a `close` method)"
```

Comment:
An exception message probably shouldn't use backquotes; they won't get rendered. Use `close()` instead, and let the `()` indicate that it's code.

Possible alternative:
```python
raise TypeError(
    "Invalid stream object for context management; "
    "%s has no close() method." % type(self.stream).__name__)
```

Reply:
obviated by 9f7d5c696e5d39c449ec9442a12e259f310fd3b9